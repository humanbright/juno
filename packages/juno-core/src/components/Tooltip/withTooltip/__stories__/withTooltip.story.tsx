import React, { forwardRef } from 'react';

import { RcTooltipProps, withTooltip } from '@ringcentral/juno';
import { Meta, Story } from '@storybook/react';

export default {
  title: 'ðŸš€ Cleanup Components/Tooltip/withTooltip',
  argTypes: {
    useRcTooltip: {
      control: {
        type: 'boolean',
      },
    },
    placement: {
      options: [
        'top',
        'top-start',
        'top-end',
        'right',
        'right-start',
        'right-end',
        'bottom',
        'bottom-start',
        'bottom-end',
        'left',
        'left-start',
        'left-end',
      ],
      control: {
        type: 'select',
      },
    },
    title: {
      control: {
        type: 'text',
      },
    },
  },
} as Meta;

type CustomComponentProps = { content?: string };
const CustomComponent = forwardRef<HTMLButtonElement, CustomComponentProps>(
  ({ content, ...rest }, ref) => {
    return (
      <button {...rest} ref={ref}>
        {content}
      </button>
    );
  },
);

const WithTooltipComponent = withTooltip(CustomComponent);
type WithTooltipProps = {
  placement: RcTooltipProps['placement'];
  title: string;
  useRcTooltip: boolean;
};

export const _withTooltip: Story<WithTooltipProps> = ({
  placement,
  ...rest
}) => {
  return (
    <WithTooltipComponent
      content="Custom Button"
      {...rest}
      TooltipProps={{
        placement,
      }}
    />
  );
};

_withTooltip.args = {
  title: 'button',
  useRcTooltip: true,
  placement: 'bottom',
};

_withTooltip.storyName = 'withTooltip';
