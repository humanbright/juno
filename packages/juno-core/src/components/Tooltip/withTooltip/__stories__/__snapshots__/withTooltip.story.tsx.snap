// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots ðŸš€ Cleanup Components/Tooltip/withTooltip withTooltip 1`] = `
<button
  aria-describedby={null}
  className=""
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  title="button"
>
  Custom Button
</button>
`;
